codegen-winter/src/air/boundary_constraints.rs

```diff
@@ -55,16 +55,19 @@ trait Codegen {
 impl Codegen for Expr {
     fn to_string(&self) -> String {
         match self {
-            Self::Constant(value) => format!("Felt::new({})", value),
+            Self::Const(value) => format!("Felt::new({})", value),
             Self::Add(lhs, rhs) => {
                 format!("{} + {}", lhs.to_string(), rhs.to_string())
             }
-            Self::Subtract(lhs, rhs) => {
+            Self::Sub(lhs, rhs) => {
                 format!("{} - {}", lhs.to_string(), rhs.to_string())
             }
-            Self::Multiply(lhs, rhs) => {
+            Self::Mul(lhs, rhs) => {
                 format!("{} * {}", lhs.to_string(), rhs.to_string())
             }
+            Self::Exp(lhs, rhs) => {
+                format!("({}).exp({})", lhs.to_string(), rhs)
+            }
```

Comment:
I think parentheses need to be added to the other expressions above like in the `Mul` PR. These can be revisited at the same time those ones are

Reply:
Actually, since this was a pre-existing issue I'll fix it separately and approve this PR